File: app/src/main/java/org/digitalcampus/oppia/model/ActivityCount.java
package org.digitalcampus.oppia.model;

import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

public class ActivityCount {

    private Map<String, Integer> typeCount = new LinkedHashMap<>();

    public static ActivityCount initialize(List<ActivityType> activityTypes) {
        ActivityCount activityCount = new ActivityCount();
        for (ActivityType activityType : activityTypes) {
            activityCount.getTypeCount().put(activityType.getType(), 0);
        }
        return activityCount;
    }

    public Map<String, Integer> getTypeCount() {
        return typeCount;
    }

    public void setTypeCount(Map<String, Integer> typeCount) {
        this.typeCount = typeCount;
    }

    public void incrementNumberActivityType(String type) {
        int incrementedActivitiesNumber = typeCount.get(type) + 1;
        typeCount.put(type, incrementedActivitiesNumber);
    }

    public int getValueForType(String type) {
        Integer value = typeCount.get(type);
        return value != null ? value : 0;
    }

    public boolean hasValidEvent(String event) {
        return typeCount.containsKey(event);
    }
}

File: app/src/main/java/org/digitalcampus/oppia/model/ActivityCount.kt
package org.digitalcampus.oppia.model

class ActivityCount {
    var typeCount: MutableMap<String, Int> = LinkedHashMap()

    companion object {
        @JvmStatic
        fun initialize(activityTypes: List<ActivityType>): ActivityCount {
            val activityCount = ActivityCount()
            for (activityType in activityTypes) {
                activityCount.typeCount[activityType.type] = 0
            }
            return activityCount
        }
    }


    fun incrementNumberActivityType(type: String) {
        typeCount[type] = typeCount.getOrDefault(type, 0) + 1
    }

    fun getValueForType(type: String): Int {
        return typeCount[type] ?: 0
    }

    fun hasValidEvent(event: String): Boolean {
        return typeCount.containsKey(event)
    }

}